bin/init.js

```diff
@@ -1,188 +1,317 @@
-#!/usr/bin/env node
-
-const path = require( 'path' );
-const fs = require( 'fs' );
-const prompt = require( 'prompt-sync' )();
-const replace = require( 'replace-in-file' );
-const rootDir = path.join( __dirname, '..' );
-
-// Helpers
-const fgRed = '\x1b[31m';
-const fgGreen = '\x1b[32m';
-const fgBlue = '\x1b[34m';
-const fgMagenta = '\x1b[35m';
-const fgCyan = '\x1b[36m';
-
-// Functions
-const consoleOutput = ( color, text ) => {
-	console.log( color, text );
+#! /usr/bin/env node
+
+/* eslint no-console: 0 */
+
+/**
+ * External dependencies
+ */
+const fs = require('fs');
+const path = require('path');
+const readline = require('readline');
+
+/**
+ * Define Constants
+ */
+const rl = readline.createInterface({
+	input: process.stdin,
+	output: process.stdout,
+});
+const info = {
+	error: (message) => {
+		return `\x1b[31m${ message }\x1b[0m`;
+	},
+	success: (message) => {
+		return `\x1b[32m${ message }\x1b[0m`;
+	},
+	warning: (message) => {
+		return `\x1b[33m${ message }\x1b[0m`;
+	},
+	message: (message) => {
+		return `\x1b[34m${ message }\x1b[0m`;
+	},
 };
+let fileContentUpdated = false;
+let fileNameUpdated = false;
+
+// Start with a prompt.
+rl.question('Would you like to setup the theme? (Y/n) ', (answer) => {
+	if ('n' === answer.toLowerCase()) {
+		console.log(info.warning('\nTheme Setup Cancelled.\n'));
+		process.exit(0);
+	}
+	rl.question('Enter theme name (shown in WordPress admin)*: ', (themeName) => {
+		const themeInfo = setupTheme(themeName);
+		rl.question('Confirm the Theme Details (Y/n) ', (confirm) => {
+			if ('n' === confirm.toLowerCase()) {
+				console.log(info.warning('\nTheme Setup Cancelled.\n'));
+				process.exit(0);
+			}
+			initTheme(themeInfo);
+			rl.close();
+		});
+	});
+});
+
+rl.on('close', () => {
+	process.exit(0);
+});
+
+/**
+ * Theme Setup
+ *
+ * @param {string} themeName
+ *
+ * @return {Object} themeInfo
+ */
+const setupTheme = (themeName) => {
```

Comment:
```suggestion
/**
 * Renders the theme setup modal with all necessary information related to the search-replace.
 *
 * @param {string} themeName
 *
 * @return {Object} themeInfo
 */
const loadThemeSetupModal = ( themeName ) => {
```
Thoughts?

Reply:
We can use it, but to be more precise this function generates a box of theme details not a Modal. 
We can also name it like - `setupThemeDetails`.

Reply:
Noted, go ahead with `setupThemeDetails` or `renderThemeDetails`.